### PR TITLE
[Godot 4] Fix Popup Bugs

### PR DESCRIPTION
**Pull request by pcen**
Incrementally removing bugs from the godot4 branch:

- prevents polling for input actions in Game.cs when a popup is visible
- fixes behaviour when the build city popup is open and esc. is pressed: before it would close the popup and show the exit game popup at the same time. now, the first esc. press will only close the current popup

### CODE REVIEW

File: C7/Game.cs [hunk 1]
```diff
@@ -440,6 +442,15 @@ enum GameState {
 
 	// Handle Godot keybind actions
 	private void processActions() {
+		if (Input.IsActionJustPressed(C7Action.Escape) && popupOverlay.Visible) {
+			popupOverlay.OnHidePopup();
+			return;
+		}
+
+		// never poll for actions if UI elements are visible
+		if (popupOverlay.Visible) {
```
Comment (WildWeazel) [inline C7/Game.cs]: what's the distinction between Visible and ShowingPopup()?

Comment (pcen) [reply C7/Game.cs]: hmm I wrote ShowingPopup to use here but I must have forgotten to change it... currently there is no difference aside from PopupOverlay.ShowingPopup signalling clear intent vs. the overlay itself being visible which is less clear imo. Will fix

Comment (WildWeazel) [reply C7/Game.cs]: bump

**Comment by pcen** [comment]
discovered https://github.com/C7-Game/Prototype/issues/424 while cleaning up this change

**Comment by Sean-Brown** [comment]
Getting back into this and not very familiar with all the SDKs involved, but the code changes still make sense to me.

**Review by QuintillusCFC — APPROVED** [review]

**Comment by QuintillusCFC** [comment]
Reviewed code, the changes make sense, and should prevent the multi-popup issue generally.  Approved and merging.

**Pull request merged**